pkgs/development/python-modules/soco/default.nix

```diff
@@ -15,42 +18,55 @@
 
 buildPythonPackage rec {
   pname = "soco";
-  version = "0.20";
+  version = "0.21.2";
+  disabled = isPy27;
 
   # N.B. We fetch from GitHub because the PyPI tarball doesn't contain the
   # required files to run the tests.
   src = fetchFromGitHub {
     owner = "SoCo";
     repo = "SoCo";
     rev = "v${version}";
-    sha256 = "0p87aw7wxgdjz0m0nqqcfvbn24hlbq1hh1zxdq2c0k2jcbmaj8zc";
+    sha256 = "sha256-CCgkzUkt9YqTJt9tPBLmYXW6ZuRoMDd7xahYmNXgfM0=";
   };
 
+  patches = [(fetchpatch {
+    url = "https://patch-diff.githubusercontent.com/raw/SoCo/SoCo/pull/811.patch";
+    sha256 = "sha256-GBd74c8zc25ROO411SZ9TTa+bi8yXJaaOQqY9FM1qj4=";
+  })];
+
   # N.B. These exist because:
   # 1. Upstream's pinning isn't well maintained, leaving dependency versions no
   #    longer in nixpkgs.
   # 2. There is no benefit for us to be running linting and coverage tests.
   postPatch = ''
     sed -i "/black/d" ./requirements-dev.txt
+    sed -i "/coveralls/d" ./requirements-dev.txt
+    sed -i "/flake8/d" ./requirements-dev.txt
     sed -i "/pylint/d" ./requirements-dev.txt
     sed -i "/pytest-cov/d" ./requirements-dev.txt
   '';
 
   propagatedBuildInputs = [
+    ifaddr
     requests
     toml
     xmltodict
   ];
+
   checkInputs = [
     pytestCheckHook
-    coveralls
-    flake8
     graphviz
     mock
+    requests-mock
     sphinx
     sphinx_rtd_theme
   ];
 
+  passthru.updateScript = nix-update-script {
+    attrPath = "python3Packages.${pname}";
+  };
```

Comment:
I would prefer if people would just use ``nix-update python3Packages.soco`` instead of adding those 3 lines to everything.